iup/eval_projections.py

```diff
@@ -1,5 +1,86 @@
+import polars as pl
 import altair as alt
 
+# get mspe #
+def get_mspe(
+        data_df,
+        pred_df,
+        var
+):
+    """
+    Calculate the mean squared prediction error between the observed
+    data and prediction when both are available. 
+
+    Parameters:
+    --------------
+    data_df: polars.DataFrame
+        The observed vaccine uptake. Must include `date` and the variable-
+        of-interest (var).
+    pred_df: polars.DataFrame
+        The predicted daily uptake. Must include `date` and the variable-
+        of-interest (var).
+    var: string
+        The name of the variable-of-interest.
+    
+    Return:
+    -------------
+    A polars.DataFrame with the MSPE and the date of initializing forecast. 
+    
+    """
+    
+    
+    mspe = data_df.join(
+            pred_df, 
+            on = 'date',
+            how = 'inner'
+        ).with_columns(
+            spe = (pl.col(var) - pl.col(f"{var}_right"))**2
+        ).with_columns(
+            mspe = pl.col('spe').mean(),
+        ).filter(
+            pl.col('date') == pl.col('date').min(),
+        ).select(
+            'date','mspe'
+        )
+    
+    return mspe
+
+
+# get end-of-season total uptake #
+
+def get_eos(
```

Comment:
I would have expected that this function would have the same signature as the one before it. We'll want all our evaluation functions to have the same signature

Reply:
also, this doesn't seem like a score function, per se? it just pulls out a particular data point

Reply:
Right, this function just pulls out the total uptake at the end of the season. I can make this function to calculate the difference in the total uptake between the predicted and the observed data, so it will have the same signature.

Reply:
right, like I think you want this to be something like absolute error at end of season (percentage points? or % error?)

Reply:
Sounds good. In next PR incorporating the OOP, I will add the absolute error relative to the observed end-of-season uptake: like abs(data-predicted)/data to see how deviated the prediction is compared to the data.